org.eclipse.winery.repository.rest/src/main/java/org/eclipse/winery/repository/rest/resources/threats/ThreatsResource.java

```diff
@@ -0,0 +1,84 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache Software License 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ *******************************************************************************/
+package org.eclipse.winery.repository.rest.resources.threats;
+
+import java.io.IOException;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.eclipse.winery.model.threatmodeling.ThreatModelingUtils;
+import org.eclipse.winery.repository.backend.IRepository;
+import org.eclipse.winery.repository.backend.RepositoryFactory;
+
+import com.sun.jersey.multipart.FormDataParam;
+
+public class ThreatsResource {
+    
+    // reuse repository
+    private IRepository repository = RepositoryFactory.getRepository();
+
+    /**
+     * generate list of all threat policy templates in the repository
+     */
+    @Produces(MediaType.APPLICATION_JSON)
+    @GET
+    public Response threatCatalogue() {
+        ThreatModelingUtils tmu = new ThreatModelingUtils(this.repository);
+        return Response.ok(tmu.getThreatCatalogue()).build();
```

Comment:
You don't need to create the response manually, just use `return tmu.getThreatCataloge();`. Jackson & Jersy will handle everything else for you automatically.

Reply:
fixed